peeklock/renewer.go

```diff
@@ -0,0 +1,54 @@
+package peeklock
+
+import (
+	"context"
+	"time"
+
+	servicebus "github.com/Azure/azure-service-bus-go"
+)
+
+type LockRenewer interface {
+	RenewLocks(ctx context.Context, messages ...*servicebus.Message) error
+}
+
+type PeriodicLockRenewer struct {
+	lockRenewer LockRenewer
+	cancelFunc  func()
+	timer       *time.Timer
+}
+
+func RenewPeriodically(ctx context.Context, interval time.Duration, lockrenewer LockRenewer, msg *servicebus.Message) *PeriodicLockRenewer {
+	p := &PeriodicLockRenewer{
+		lockRenewer: lockrenewer,
+	}
+	go p.startPeriodicRenewal(ctx, interval, msg)
+	return p
+}
+
+func (plr *PeriodicLockRenewer) startPeriodicRenewal(ctx context.Context, interval time.Duration, message *servicebus.Message) {
+	var renewerCtx context.Context
+	renewerCtx, plr.cancelFunc = context.WithCancel(ctx)
+	for alive := true; alive; {
+		select {
+		case <-time.After(interval):
+			// add span for logs
```

Comment:
would u like to uncomment the logs? :D

Reply:
No that's don't print. I added an issue to move the tracing helper I to a package, then we can trace everywhere. That was for test. I'll delete the commented logs